blocks/data.py

```diff
@@ -8,29 +8,37 @@
 
 from .base import DataSource
 from .pipeline import Pipeline, process_block
+from .base import Element
 
 
 class BaseConcat(DataSource):
 
     blocks: List[Union[DataSource, Pipeline]]
 
-    def __init__(self, blocks: List[Union[DataSource, Pipeline]]):
+    def __init__(self, id: str, blocks: List[Union[DataSource, Pipeline]]):
         self.blocks = blocks
+        self.id = id
 
-    def deplate(self, store: Store) -> pd.DataFrame:
+    def deplate(self, store: Store, plugins: List["Plugin"]) -> pd.DataFrame:
         collected = self.transform(
-            [process_block(block, store) for block in self.blocks]
+            [process_block(block, store, plugins) for block in self.blocks]
         )
 
         return collected
 
-    def load(self):
+    def load(self, plugins: List["Plugin"]):
         for block in self.blocks:
-            block.load()
+            block.load(plugins)
 
     def transform(self, data: List[pd.Series]) -> pd.Series:
         raise NotImplementedError()
 
+    def children(self) -> List[Element]:
+        return [self] + [self.blocks]
```

Comment:
I think if you use this, that should be OK? (same with Ensemble) ```        return [self] + [block.children() for block in self.blocks]```

Reply:
or is it not:
``return [self] + [self.blocks] + [block.children() for block in self.blocks]``
?

Reply:
the pipelines will already give themselves back inside children right now, so you don't need that!